test/e2e/playwright/shared/pageObjects/network-controller-page.ts

```diff
@@ -72,13 +96,38 @@ export class NetworkController {
     const addBtn = this.page.getByTestId(`add-network-${options.networkName}`);
     await addBtn.click();
     await this.approveBtn.click();
-    await this.switchToNetworkBtn.click();
     await this.gotItBtn.click();
   }
 
-  async selectNetwork(options: { networkName: string }) {
-    await this.networkDisplay.click();
-    await this.networkSearch.fill(options.networkName);
-    await this.page.getByText(options.networkName).click();
+  async selectNetwork(options: {
+    name: string;
+    rpcName: string;
+    url: string;
+    chainID: string;
+    symbol: string;
+  }) {
+    const currentNetwork = await this.networkDisplay.textContent();
+    if (currentNetwork !== options.name) {
+      await this.networkDisplay.click();
+      if (options.name === Tenderly.Mainnet.name) {
+        await this.page.getByText(options.rpcName).click();
+        await this.page.getByText(options.rpcName).click();
+      } else {
+        await this.page.getByTestId(options.name).click();
+      }
+      await this.waitForNetworkToSwitch(options.name);
+      await this.page.waitForTimeout(5000);
+    }
+  }
+
+  async waitForNetworkToSwitch(networkName: string) {
```

Comment:
I think we need to improve this function, because  the loop will continue indefinitely if the network never switches to the expected one, which could lead to an infinite loop.

Reply:
I have removed the functions as it's not needed with the new network UI

Reply:
hmm... I still see this function in the file changes.

Reply:
Sorry wrong comment I didn't remove the function, just removed the loop

Reply:
thanks !